crates/bevy_ecs/src/query/access.rs

```diff
@@ -59,39 +61,56 @@ impl<T: SparseSetIndex> Access<T> {
         self.writes.contains(index.sparse_set_index())
     }
 
-    /// Sets this `Access` to having read access for all indices.
+    /// Sets this `Access` as having read access to all indexed elements.
     pub fn read_all(&mut self) {
         self.reads_all = true;
     }
 
-    /// Returns true if this `Access` has read access to all indices.
+    /// Returns `true` if this `Access` has read access to all indexed elements.
+    #[inline]
     pub fn reads_all(&self) -> bool {
         self.reads_all
     }
 
-    /// Clears all recorded accesses.
+    /// Sets this `Access` as having write access to all indexed elements.
+    pub fn write_all(&mut self) {
+        self.reads_all = true;
+        self.writes_all = true;
+    }
+
+    /// Returns `true` if this `Access` has write access to all indexed elements.
+    #[inline]
+    pub fn writes_all(&self) -> bool {
+        self.writes_all
+    }
+
+    /// Removes all accesses.
     pub fn clear(&mut self) {
         self.reads_all = false;
+        self.writes_all = false;
         self.reads_and_writes.clear();
         self.writes.clear();
     }
 
     /// Extends this `Access` with another, copying all accesses of `other` into this.
     pub fn extend(&mut self, other: &Access<T>) {
         self.reads_all = self.reads_all || other.reads_all;
+        self.writes_all = self.writes_all || other.writes_all;
         self.reads_and_writes.union_with(&other.reads_and_writes);
         self.writes.union_with(&other.writes);
     }
 
     /// Returns true if this `Access` is compatible with `other`.
     ///
-    /// Two `Access` instances are incompatible with each other if one `Access` has a write for
-    /// which the other also has a write or a read.
+    /// Two `Access` instances are incompatible if one writes
+    /// an index that the other writes or reads.
     pub fn is_compatible(&self, other: &Access<T>) -> bool {
-        if self.reads_all {
-            0 == other.writes.count_ones(..)
+        if self.writes_all || other.writes_all {
```

Comment:
This is incorrect if other access has no reads or writes and vis versa.

Reply:
Would an example be a system with `()` as its only parameter?

Reply:
Ok, I think I've corrected this.

Reply:
looks good.

My concern would be more for if this API gets reused somewhere else rather than just for system scheduling or checking the current system params.